packages/frontend/components/auth/sign-up/component.tsx

```diff
@@ -0,0 +1,107 @@
+import { Button } from '@primitives';
+import type { SubmitHandler } from 'react-hook-form';
+import { useForm } from 'react-hook-form';
+import type { UserSignUpDto } from '@vse-bude/shared';
+import { useTranslation } from 'next-i18next';
+import { joiResolver } from '@hookform/resolvers/joi';
+import { useAppDispatch, useTypedSelector } from '@hooks';
+import { Input, PasswordInput } from '../../primitives/input';
+import { form, inputWrapper } from '../layout/styles';
+import { getErrorKey } from '../../../helpers/validation';
+import { signUpUser } from '../../../store/auth';
+import { Loader } from '../../primitives/loader';
+import { signUpSchema } from './validation';
+
+export const SignUpForm = () => {
+  const { t: lang } = useTranslation('validation');
+  const { t: commonLang } = useTranslation('common');
```

Comment:
const { t } = useTranslation(['common', 'validation']);
...
t('common:key')

Reply:
I think, with such approach as now it will be better and easier to read.
Because we need do next with you advice:
`errors.firstName?.type && t(`validation:${getErrorKey('firstName', errors.firstName?.type)}`)`